### DISCUSSION TITLE
Can't deploy or install

### DISCUSSION THREAD

**Discussion by mnfctrdinitaly** [HiEventsDev/Hi.Events#37] [Q&A]
### Help!
I have now installed this four different ways. In each version I get the same error message. There must be something I am doing wrong...

### 1. Deploy on Digital Ocean? 
Front end loads, but upon registering it gives an "unexpected error"

### 2. Deploy on Railway?
Front end loads, but upon registering it gives an "unexpected error"

### 3. Deploy on a working docker/docker-compose host?
Front end loads, but upon registering it gives an "unexpected error"

### 4. Deploy locally on a fresh debian install?
Front end loads, but upon registering it gives an "unexpected error"

Is there some .env or config file that I am just struggling or forgetting or not knowing how to set? I have followed the guides step by step and (admittedly they sometimes are broken or link to 404 not found, so they aren't always the most helpful.)

Can someone please tell what I am doing wrong?

**Comment by mnfctrdinitaly** [comment]
Okay, I have made ONE of the configurations work:

If I deploy on a fresh install debian the all-in-one using the instruction in the README.md (not in the documentation wiki) and THEN edit the docker-compose.yml to update the env variable from "localhost:port" to the actual public IP address of the host then it works... 

but how do I get this working behind a reverse proxy without having to any ports beyond the two port I have open to nginx that then proxies everything to my various containers?

When I use a working site.conf for another container on the same host and just update the proxy_pass container to "all-in-one" which is the default name of the container I can get to the login page but nothing beyond that and the console says

```
[Error] Origin https://sub.domain.tld is not allowed by Access-Control-Allow-Origin.
[Error] XMLHttpRequest cannot load https://sub.domain.tld/api/users/me due to access control checks.
[Error] Failed to load resource: Origin https://sub.domain.tld is not allowed by Access-Control-Allow-Origin. (me, line 0)
```

**Comment by daveearley** [comment]
Hi @mnfctrdinitaly 👋

Sorry you're having trouble. I tried the DigitalOcean install and it appears to work as expected.

Did you fill in values for these fields while deploying?

<img width="869" alt="Screenshot 2024-06-15 at 11 20 05" src="https://github.com/HiEventsDev/hi.events/assets/166798/4415fc9d-a354-40b1-91f2-a8c294676b54">

Also, did you see any errors in the logs in DigitalOcean?

Thanks,
Dave

**Comment by mnfctrdinitaly** [comment]
I got it working... I used the all-in-one docker-compose.yml and updated the env variables.

Are there issues or things I should fix in this?

```yaml
    environment:
      - VITE_FRONTEND_URL=https://subdomain.domain.tld
      - VITE_API_URL_CLIENT=https://subdomain.domain.tld/api
      - VITE_API_URL_SERVER=https://subdomain.domain.tld/api
      - VITE_STRIPE_PUBLISHABLE_KEY=pk_test
      - LOG_CHANNEL=stderr
      - QUEUE_CONNECTION=sync
      - MAIL_MAILER=array
      - APP_KEY=secret
      - JWT_SECRET=secret
      - FILESYSTEM_PUBLIC_DISK=public
      - FILESYSTEM_PRIVATE_DISK=local
      - APP_DEBUG=TRUE
      - APP_CDN_URL=https://subdomain.domain.tld/storage
      - DATABASE_URL=postgresql://postgres:secret@postgres:5432/hi-events
      - APP_DISABLE_REGISTRATION=true
      - APP_FRONTEND_URL=https://subdomain.domain.tld
      - MAIL_MAILER=smtp
      - MAIL_HOST=smtp.gmail.com
      - MAIL_PORT=587
      - MAIL_USERNAME=secret
      - MAIL_PASSWORD=secret
      - STRIPE_PUBLIC_KEY=secret
      - STRIPE_SECRET_KEY=secret
      - STRIPE_WEBHOOK_SECRET=secret
```

**Reply by FourHeadedDev** [reply]
am facing the same issue do yous got fixed?
